src/main/java/core/basesyntax/abstracts/Figure.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.abstracts;
+
+import core.basesyntax.interfaces.Calculatable;
+import core.basesyntax.interfaces.Drawable;
+
+import java.text.DecimalFormat;
+
+public abstract class Figure implements Drawable, Calculatable {
+    private String color;
+    private final DecimalFormat decimalFormat = new DecimalFormat( "#.#" );
+
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public String setDecimalFormat(double number) {
+        return decimalFormat.format(number);
+    }
```

Comment:
1. You have class variable decimalFormat and method setDecimalFormat which is obviously should be setter. But it's not, it even doesn't set anything, it returns formatted value as String. This makes naming absolutely confusing
2. You use this method inside child classes, why public then?
```suggestion
    protected String getFormattedDecimal(double number) {
        return decimalFormat.format(number);
    }
```

Reply:
probably most stupidest thing I ever done, I'm really very sorry you should read this :(